### PR TITLE
hometask-3/rainfall

### PR DESCRIPTION
**Pull request by NastassiaBon**

### CODE REVIEW

File: NastassiaBon/hometask-3/rainfall.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+def data_extraction
+  file = File.new('data.txt')
+  file.read
+end
+
+def mean(town, data)
+  return -1.0 unless data.include? town
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb]: Better use brackets if the method takes arguments - `data.include?(town)`. You can not use brackets for `p`, `puts`, etc., or methods w/o arguments.

File: NastassiaBon/hometask-3/rainfall.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  serch_town(town, data).sum / 12
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb]: What is 12? Cat pictures in my pocket? Use constants for that type of variable.

File: NastassiaBon/hometask-3/rainfall.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+end
+
+def variance(town, data)
+  return -1.0 unless data.include? town
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb]: Brackets

File: NastassiaBon/hometask-3/rainfall.rb (hunk 1 above)
Comment (Kaddyluck) [reply NastassiaBon/hometask-3/rainfall.rb]: Also, why -1.0? Why not -1? I need explanation :)

File: NastassiaBon/hometask-3/rainfall.rb (hunk 3 above)
Comment (Kaddyluck) [reply NastassiaBon/hometask-3/rainfall.rb]: Why float -1?

File: NastassiaBon/hometask-3/rainfall.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  serch_town(town, data).map { |i| (mean(town, data) - i)**2 }.sum / 12
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb]: 12 cat pictures? 12 miles per bear?

Comment (Kaddyluck) [reply NastassiaBon/hometask-3/rainfall.rb]: Maybe se**A**rch_town?

File: NastassiaBon/hometask-3/rainfall.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+end
+
+def serch_town(town, data)
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb]: Maybe se**A**rch_town?

File: NastassiaBon/hometask-3/rainfall.rb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  data.split("\n").each do |i|
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb]: Better not to use one-letter variables. What is `i` in your data? City? So rename it into the `city`

File: NastassiaBon/hometask-3/rainfall.rb [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    if i.include? town
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb]: Brackets

File: NastassiaBon/hometask-3/rainfall.rb [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      string = i.scan(/\d*\.\d/).map(&:to_f)
+      return string
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb]: You don't need to return the string. Ruby already returning the value from the last line
image

File: NastassiaBon/hometask-3/rainfall.rb [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    end
+  end
+end
+
+def run_cli
+  loop do
+    puts 'Enter city name: '
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb]: Move to constant, pls

File: NastassiaBon/hometask-3/rainfall.rb [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    string = gets.chomp
+    break if string == 'exit'
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb]: Move to constant, pls

File: NastassiaBon/hometask-3/rainfall.rb [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+    string.empty? ? puts('City name can not be blank!') : show_result(string)
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb]: Move string to constant, pls

File: NastassiaBon/hometask-3/rainfall.rb [hunk 12]
```diff
@@ -1,26 +1,31 @@
 #!/usr/bin/env ruby
 # frozen_string_literal: true
 
+ENTER_MESSAGE = 'Enter city name: '
+ERROR_MESSAGE = 'City name can not be blank!'
+EXIT_MESSAGE = 'exit'
+MONTH = 12
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb]: MONTH - 1, NUMBER_OF_MONTHS - 12

File: NastassiaBon/hometask-3/rainfall.rb [hunk 13]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+ENTER_MESSAGE = 'Enter city name: '
+ERROR_MESSAGE = 'City name can not be blank!'
+EXIT_MESSAGE = 'exit'
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb]: `Если пользователь ввел exit!, программа должна завершиться.`

File: NastassiaBon/hometask-3/rainfall.rb (hunk 8 above)
Comment (Kaddyluck) [reply NastassiaBon/hometask-3/rainfall.rb]: My bad, you need

File: NastassiaBon/hometask-3/rainfall.rb (hunk 3 above)
Comment (NastassiaBon) [reply NastassiaBon/hometask-3/rainfall.rb]: According to the task on the codewars, we are given the opportunity to choose -1 or -1.0, but both options will be correct and will not greatly affect the task

File: NastassiaBon/hometask-3/rainfall.rb [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+end
+
+def variance(town, data)
+  return -1 unless data.include?(town)
+
+  search_town(town, data).map { |city| (mean(town, data) - city)**2 }.sum / NUMBER_OF_MONTHS
+end
+
+def search_town(town, data)
+  data.split("\n").each do |city|
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb:27]: Sorry for the previous comment, change city to smth like `line`. Use the name of the entity we iterate over

File: NastassiaBon/hometask-3/rainfall.rb [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+def mean(town, data)
+  return -1 unless data.include?(town)
+
+  search_town(town, data).sum / NUMBER_OF_MONTHS
+end
+
+def variance(town, data)
+  return -1 unless data.include?(town)
+
+  search_town(town, data).map { |city| (mean(town, data) - city)**2 }.sum / NUMBER_OF_MONTHS
```
Comment (Kaddyluck) [inline NastassiaBon/hometask-3/rainfall.rb:23]: Change `city` to iteratable entity, smth like 'value`

**Review by Kaddyluck — APPROVED** [review]

**Pull request merged**